shcomplete/repos.py

```diff
@@ -0,0 +1,78 @@
+import time
+import os
+import argparse
+import sys
+
+from github import Github as GitHub, RateLimitExceededException
+
+class GithubShell(object):
+    """
+    Collects the list of repositories with either bash/zsh/fish history files
+    """
+
+    def __init__(self, login_or_token, password=None, **kwargs):
+        kwargs.setdefault("per_page", 30)
+        self._api = GitHub(login_or_token, password=password, **kwargs, timeout=40)
+
+    def get_repos_list(self):
+        """
+        Get the list of repositories
+        """
+
+        repos_fish = []
+        for num, repo in enumerate(self._api.search_code("filename:fish_history size:>0")):
+            repos_fish.append(repo.html_url)
+            print(num + 1, repos_fish[-1])
+            time.sleep(0.5)
+        repos_fish = {x for x in repos_fish if x.endswith("/fish_history")}
+
+        repos_bash = []
+        for num, repo in enumerate(self._api.search_code("filename:bash_history size:>=950 -user:Dahs81")):
+            repos_bash.append(repo.html_url)
+            print(num + 1, repos_bash[-1])
+            time.sleep(0.7)
+        repos_bash = {x for x in repos_bash if x.endswith("/bash_history") or x.endswith("/.bash_history")}
+
+        repos_zsh = []
+        for num, repo in enumerate(self._api.search_code("filename:zsh_history -extension:zsh size:>0")):
+            repos_zsh.append(repo.html_url)
+            print(num + 1, repos_zsh[-1])
+            time.sleep(0.7)
+        repos_zsh = {x for x in repos_zsh if x.endswith("/zsh_history") or x.endswith("/.zsh_history")}
+
+        repos = repos_fish.copy()
+        repos.update(repos_bash)
+        repos.update(repos_zsh)
+
+        return repos
+
+    def to_raw_urls(self, list_urls):
+        """
+        Transform the links returned by GitHub API, github.com into raw.githubusercontent.com
+        """
+
+        list_raw_urls = []
+        base = "https://raw.githubusercontent.com"
+        for url in list_urls:
+            url = url.split("/")
+            raw_url = base
+            for i in range(3, len(url)):
+                if url[i] != "blob":
+                    raw_url += "/" + url[i]
+            list_raw_urls.append(raw_url)
+
+        return list_raw_urls
+
+
+def fetch_repos(args):
+    """
+    Fetch the repositories and write the raw links into an output txt file
+    """
+
+    histories = GithubShell(args.token)
+    list_repos = histories.get_repos_list()
+    list_repos = histories.to_raw_urls(list_repos)
+    with open(args.output, 'w') as f:
+        ens_repos = set(list_repos)
+        for rep in list(ens_repos):
```

Comment:
You do not need to convert it to list if you just want to iterate. If you want the same order each time, use `sorted`